### DISCUSSION TITLE
Image

### DISCUSSION THREAD

**Discussion by wartburggraaf** [nl-design-system/backlog#330] [Component suggestions]
<img width="480" alt="visuele weergave van een Image component" src="https://github.com/user-attachments/assets/75e6e153-32fb-4d81-8a8c-b41ab5a3db59">

We willen graag meer leren over de Image component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1725443136449139)

## Naam

Image

## Doel

Toont een afbeelding.

## Gerelateerde componenten

[Figure](https://github.com/orgs/nl-design-system/discussions/221)

## Onderzoek

[Responsive images - MDN docs](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images)
Artikel bij MDN docs over het gebruik van responsive afbeeldingen.

[Aspect ratio - MDN docs](https://developer.mozilla.org/en-US/docs/Web/CSS/aspect-ratio)
Documentatie bij MDN docs over het gebruik van de aspect ratio CSS property.

[Object fit - MDN docs](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit)
Documentatie bij MDN docs over het gebruik van de object fit CSS property.

[Alt-texts: The Ultimate Guide](https://axesslab.com/alt-texts/)
Uitgebreide blogpost over het toepassen en schrijven van alt teksten.

[Images - GOV.UK](https://design-system.service.gov.uk/styles/images/)
Documentatie van GOV.UK rondom het gebruik van afbeeldingen. [Bijbehordende GitHub discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/70). 

## Notities
HTML richtlijnen
- Gebruik het `alt` attribuut voor een beschrijvende tekst.
- Gebruik niet het `title` attribuut voor `<img>`.
  - De afbeelding moet goed zichtbaar zijn, maar sommige browsers maken een tooltip met voor de tekst in het title attribuut.
  - Een tooltip kan bijvoorbeeld een afbeelding van een [QR-code](https://en.wikipedia.org/wiki/QR_code) onbruikbaar maken doordat de QR-code niet meer herkend wordt.

[Using the HTML title attribute — Steve Faulkner](https://www.tpgi.com/using-the-html-title-attribute-updated/)

## Relevante WCAG richtlijnen
- [WCAG eis 1.1.1](https://www.w3.org/TR/WCAG21/#non-text-content): afbeeldingen moeten een tekst als alternatief hebben, bijvoorbeeld via het alt-attribuut
- [WCAG eis 1.4.5](https://www.w3.org/TR/WCAG21/#images-of-text): gebruik tekst in plaats van afbeeldingen van tekst.
- [WCAG eis 1.4.6](https://www.w3.org/TR/WCAG21/#contrast-enhanced)
- [WCAG eis 1.4.9](https://www.w3.org/TR/WCAG21/#images-of-text-no-exception): gebruik afbeeldingen van tekst alleen als er geen alternatief is.
- [WCAG eis 1.4.11](https://www.w3.org/TR/WCAG21/#non-text-contrast)
- [WCAG eis 3.1.2](https://www.w3.org/TR/WCAG21/#language-of-parts): de taal van de alt tag moet instelbaar zijn

**Comment by MariekeBrouwer** [comment]
## Groningen

Screenshots van toepassing Image component voor Gemeente Groningen.

![Niet ingelogd](https://github.com/user-attachments/assets/0f690425-27cb-4678-b35a-7f5b32baaa64)
Afbeelding gebruiken wij ook wel in cards (ander component, maar misschien wel goed om even te noemen :) )

![Scherm_afbeelding 2024-01-30 om 09 59](https://github.com/user-attachments/assets/2c7f43e5-a592-4ede-bf06-066086d708da)
Zowel op de gemeentelijke website als op de MijnOmgeving gebruiken we een header image 

![image](https://github.com/user-attachments/assets/d9bf2bbe-b173-44fa-82b6-a8df3851e5f7)
Op 'productpagina's'/detailpagina's worden images ook gebruikt voor extra toelichting bij de tekst

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-media-image--docs)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=2927-29200)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/image)

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-09-05 om 14 13 14](https://github.com/user-attachments/assets/344ceb18-3e6e-474c-bc53-b809422703a2)

**Figma**
![Scherm­afbeelding 2024-09-05 om 14 19 29](https://github.com/user-attachments/assets/3d808df5-1600-4d56-ab20-cade3147aa57)


---

### Notities
 
- Documentatie beschikbaar (Storybook).
  - Er wordt gewerkt met srcSet om browsers het optimale formaat uit te laten zoeken.
  - Er is documentatie over het gebruik van lazy loading.
- In Figma wordt verschillende Aspect ratio's aangeboden.
- Figma:
  - Image fill: ja
  - Fill location: center
- Code:
  - block-size: auto
  - max-inline-size: 100%
  - vertical-align: middle
  - Property beschikbaar voor `object-fit: cover`

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-data-display-image--docs)
- [Figma (Huidig)](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=8553-29678&t=JKR8Ao0Dh1myyZ1Y-4)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/Image)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-05 om 14 25 02](https://github.com/user-attachments/assets/88f6a2fd-58cd-4fc9-b445-9749e2b054c5)

Image With Caption Text
![Scherm­afbeelding 2024-09-05 om 14 25 29](https://github.com/user-attachments/assets/e5c1ca9f-2a38-4b37-89ab-11b5db2b741b)

Image With Download Link
![Scherm­afbeelding 2024-09-05 om 14 25 57](https://github.com/user-attachments/assets/20ca7657-13eb-4941-8a13-8d80eb969c5c)

Image Only
![Scherm­afbeelding 2024-09-05 om 14 26 23](https://github.com/user-attachments/assets/68b0532c-3d77-48bb-b41d-5ffb2a7b8ff9)

Portrait
![Scherm­afbeelding 2024-09-05 om 14 26 44](https://github.com/user-attachments/assets/c84f676f-a927-4a53-b41a-a8e789a2e0bb)

**Figma (Huidig)**

![Scherm­afbeelding 2024-09-05 om 14 28 31](https://github.com/user-attachments/assets/c9c6006b-ec20-420a-997c-6f32ec752bf9)


---

### Notities

- Documentatie beschikbaar.
- Opties om een caption en download mogelijkheid toe te voegen.
- In Figma wordt een extra aspect ratio (3:4) aangeboden.
- In de documentatie wordt ook geschreven over een Header image, uitlijnen en een Image gallery.
- Figma
  - Image fill: ja
  - Fill location: center
 - Code
   - display: block
    - height: auto
    - width: 100%

**Comment by jeffreylauwers** [comment]
## DSO

- [Documentatie website](https://www.dso-toolkit.nl/62.8.4/components/identity/image)
- [Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/docs/html-css-image--docs)
- [GitHub]({link-naar-component-in-github})

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-05 om 15 00 57](https://github.com/user-attachments/assets/a6ec8f95-5e83-4435-b056-3e0c1adc13a6)

Circle
![Scherm­afbeelding 2024-09-05 om 15 01 03](https://github.com/user-attachments/assets/f77e8731-756c-45b3-a9de-3aa6aa857d8d)

---

### Notities

- Optie om Image responsive te maken.
- Optie om Image als cirkel te tonen.
- Code
  - vertical-align: middle
  - max-inline-size: 100%

**Comment by jeffreylauwers** [comment]
## OpenGemeenten

- [Documentatie website](https://componenten.lenteveld.nl/)
- [Patternlab](https://patternlab.lenteveld.nl/index.html)

---

### Screenshots / Varianten

**Patternlab**

[Site-PageImage](https://patternlab.lenteveld.nl/patterns/01-molecules-001-Generic-Page-Components-30-Main-Top-03-OpenGemeentenSite-PageImage-OpenGemeentenSite-PageImage/01-molecules-001-Generic-Page-Components-30-Main-Top-03-OpenGemeentenSite-PageImage-OpenGemeentenSite-PageImage.html)

![Scherm­afbeelding 2024-09-05 om 16 48 41](https://github.com/user-attachments/assets/b01b4b9a-efd3-4370-8a59-c5d970cccfd5)

Media-Gallery component te vinden onder [Generic-Content-Elements](https://patternlab.lenteveld.nl/patterns/01-molecules-002-Generic-Content-Elements/index.html)

Media-Gallery
![Scherm­afbeelding 2024-09-05 om 16 50 16](https://github.com/user-attachments/assets/9bd0c5e5-eccc-4043-914e-05f615ca4462)

Media-Gallery Caption
![Scherm­afbeelding 2024-09-05 om 16 50 49](https://github.com/user-attachments/assets/b45027f2-6caa-45c3-a74f-36c075f294ed)

Media-Gallery Columns row
![Scherm­afbeelding 2024-09-05 om 16 52 19](https://github.com/user-attachments/assets/8ebed6b4-58de-4aa1-902d-7421d4a53a7d)

Media-Gallery LinkExternal Caption
![Scherm­afbeelding 2024-09-05 om 16 53 40](https://github.com/user-attachments/assets/9c80705b-ffd3-4fa3-b2d5-bc5d29f4541e)

---

### Notities

- Veel verschillende vormen van Image gebruikt. Bovenstaande screenshots zijn dus ook nog lang niet alles.
- Code
  - display: block
  - width: 100%
  - height: auto

**Comment by jeffreylauwers** [comment]
## Rotterdam

[Figma](https://www.figma.com/design/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?node-id=525-26492&t=16V5nINon6vmQL2A-4)

---

### Screenshots / Varianten

**Figma**

![Scherm­afbeelding 2024-09-05 om 17 02 00](https://github.com/user-attachments/assets/728bf502-150f-4f68-8233-8b30f18b6f5c)

![Scherm­afbeelding 2024-09-05 om 17 03 54](https://github.com/user-attachments/assets/1f92d13b-7ec4-4940-821b-9e7852b41ac0)

---

### Notities

- 3 verschillende aspect ratios.
- Content Image component biedt de mogelijkheid voor een description.
- Figma
  - Image fill: nee

**Comment by jeffreylauwers** [comment]
## RVO

Geen Image component gevonden maar voor de Hero component wordt er wel gebruik gemaakt van een afbeelding.

- [Documentatie website](https://nl-design-system.github.io/rvo/docs/components/hero/)
- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-hero--documentatie)

---

### Screenshots / Varianten

**Storybook**

![Scherm­afbeelding 2024-09-05 om 17 08 03](https://github.com/user-attachments/assets/aad27a51-37af-4f11-999b-0be9c5c4e1ff)

### Notities

- Code
  - height: 100%
  - object-fit: cover
  - width: 100%

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-img--docs)
- [Figma](https://www.figma.com/design/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?node-id=1303-6078)
- [Github](https://github.com/nl-design-system/utrecht/tree/main/components/img)

---

### Screenshots / Varianten

**Stroybook**

Default
Styling via the `.utrecht-img` class name.

![Scherm­afbeelding 2024-09-05 om 17 15 22](https://github.com/user-attachments/assets/427e1b07-bb39-4d59-91e2-38579310c9aa)

Photo
Photo © 2022 Gemeente Utrecht.

![Scherm­afbeelding 2024-09-05 om 17 16 24](https://github.com/user-attachments/assets/9f625248-c92a-4e87-a252-5fee0b165fb5)

Alt in another language

![Scherm­afbeelding 2024-09-05 om 17 16 48](https://github.com/user-attachments/assets/d1202047-84ee-4ca2-820f-e8326e96e14d)

---

### Notities

- Meerdere properties beschikbaar in Storybook. Onder andere voor hoogte, breedte, alt-text. en taal van alt-text.
- Figma
  - Image fill: ja
  - Fill location: center
- Code
  - `.utrecht-img`
    - max-height: auto
    - max-width: auto
  - `.utrecht-img--photo`
    - max-height: 100%
    - max-width: 100%

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/images)

---

### Screenshots / Varianten

**Documentatie website**

![Scherm­afbeelding 2024-09-05 om 17 27 27](https://github.com/user-attachments/assets/28c9f93a-e453-43d4-8ab8-1e6330a9445f)

---

### Notities

- Documentatie beschikbaar waarin wordt beschreven dat men een alt-text moet toevoegen en hoe de component responsive gemaakt kan worden.
- Code
  - display: block
  - height: auto
  - max-width: 100%

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Image op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=image)

**Comment by eslook** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Image op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=image)
